From: Andrew Deason <adeason@sinenomine.net>
Date: Thu, 11 Oct 2018 00:18:17 -0500
Subject: Remove automake autoconf vars

Commit 4706854f (autoconf: updates and cleanup) removed our invocation
of AM_INIT_AUTOMAKE, which defines the output variables PACKAGE and
VERSION. Several files in our build system are still referencing
@PACKAGE@ and @VERSION@, though, leaving them un-substituted. This
most easily is seen as the AFSVersion version string remaining as
"@VERSION@" when the tree is built without git, but it also affects
some packaging in the tree.

Remove references to @VERSION@ and @PACKAGE@, replacing them with
their autoconf equivalents @PACKAGE_VERSION@ and @PACKAGE_TARNAME@.

Change-Id: I6c6a09a46c4af4259009a4a60cfdaee63d6258c2
Reviewed-on: https://gerrit.openafs.org/13357
Tested-by: BuildBot <buildbot@rampaginggeek.com>
Reviewed-by: Benjamin Kaduk <kaduk@mit.edu>
(cherry picked from commit 2f2c2ce62aa17ecac3651d64c1168af926f7458b)
---
 src/config/Makefile.version-CML.in                     |  4 ++--
 src/config/Makefile.version-NOCML.in                   |  6 +++---
 src/packaging/MacOS/OpenAFS-debug.Description.plist.in |  2 +-
 src/packaging/MacOS/OpenAFS.Description.plist.in       |  2 +-
 src/packaging/MacOS/OpenAFS.info.in                    |  2 +-
 src/packaging/MacOS/buildpkg.sh.in                     |  4 ++--
 src/packaging/MacOS/pkgbuild.sh.in                     | 12 ++++++------
 src/packaging/RedHat/openafs.spec.in                   |  2 +-
 src/tests/run-tests.in                                 |  4 ++--
 9 files changed, 19 insertions(+), 19 deletions(-)

diff --git a/src/config/Makefile.version-CML.in b/src/config/Makefile.version-CML.in
index 45c4b84..0a08a92 100644
--- a/src/config/Makefile.version-CML.in
+++ b/src/config/Makefile.version-CML.in
@@ -5,8 +5,8 @@
 # License.  For details, see the LICENSE file in the top-level source
 # directory or online at http://www.openafs.org/dl/license10.html
 
-PACKAGE=@PACKAGE@
-VERSION=@VERSION@
+PACKAGE=@PACKAGE_TARNAME@
+VERSION=@PACKAGE_VERSION@
 
 AFS_component_version_number.o: AFS_component_version_number.c $(TOP_SRCDIR)/config/mkvers
 version.xml version.txt: $(TOP_SRCDIR)/config/mkvers
diff --git a/src/config/Makefile.version-NOCML.in b/src/config/Makefile.version-NOCML.in
index c65cfe9..8a11ee8 100644
--- a/src/config/Makefile.version-NOCML.in
+++ b/src/config/Makefile.version-NOCML.in
@@ -5,12 +5,12 @@
 # License.  For details, see the LICENSE file in the top-level source
 # directory or online at http://www.openafs.org/dl/license10.html
 
-PACKAGE=@PACKAGE@
+PACKAGE=@PACKAGE_TARNAME@
 
 AFS_component_version_number.o: AFS_component_version_number.c
 
 AFS_component_version_number.c: @TOP_OBJDIR@/src/config/Makefile.version
-	( VERSION=`@abs_top_srcdir@/build-tools/git-version @abs_top_srcdir@ "@VERSION@"` && \
+	( VERSION=`@abs_top_srcdir@/build-tools/git-version @abs_top_srcdir@ "@PACKAGE_VERSION@"` && \
 	if [ "$$SOURCE_DATE_EPOCH" ]; then \
 	     echo 'char cml_version_number[]="@(#)OpenAFS '$$VERSION `date -u -d "@$$SOURCE_DATE_EPOCH" +"%Y-%m-%d" 2>/dev/null || date -u -r "$$SOURCE_DATE_EPOCH" +"%Y-%m-%d"`'";'; \
 	else \
@@ -22,7 +22,7 @@ AFS_component_version_number.c: @TOP_OBJDIR@/src/config/Makefile.version
 	rm -f AFS_component_version_number.c.NEW
 
 version.xml:
-	VERSION=`@abs_top_srcdir@/build-tools/git-version @abs_top_srcdir@ "@VERSION@"` && \
+	VERSION=`@abs_top_srcdir@/build-tools/git-version @abs_top_srcdir@ "@PACKAGE_VERSION@"` && \
 	echo '<?xml version="1.0" encoding="UTF-8"?>' >version.xml && \
 	echo '<revision>' >>version.xml && \
 	echo '<revnumber>'$$VERSION'</revnumber>' >>version.xml && \
diff --git a/src/packaging/MacOS/OpenAFS-debug.Description.plist.in b/src/packaging/MacOS/OpenAFS-debug.Description.plist.in
index 2667452..dfbb4dc 100644
--- a/src/packaging/MacOS/OpenAFS-debug.Description.plist.in
+++ b/src/packaging/MacOS/OpenAFS-debug.Description.plist.in
@@ -9,6 +9,6 @@
 	<key>IFPkgDescriptionTitle</key>
 	<string>OpenAFS debug extension</string>
 	<key>IFPkgDescriptionVersion</key>
-	<string>@VERSION@</string>
+	<string>@PACKAGE_VERSION@</string>
 </dict>
 </plist>
diff --git a/src/packaging/MacOS/OpenAFS.Description.plist.in b/src/packaging/MacOS/OpenAFS.Description.plist.in
index 8dece17..3cab1f7 100644
--- a/src/packaging/MacOS/OpenAFS.Description.plist.in
+++ b/src/packaging/MacOS/OpenAFS.Description.plist.in
@@ -9,6 +9,6 @@
 	<key>IFPkgDescriptionTitle</key>
 	<string>OpenAFS</string>
 	<key>IFPkgDescriptionVersion</key>
-	<string>@VERSION@</string>
+	<string>@PACKAGE_VERSION@</string>
 </dict>
 </plist>
diff --git a/src/packaging/MacOS/OpenAFS.info.in b/src/packaging/MacOS/OpenAFS.info.in
index a3d04a5..ac40126 100644
--- a/src/packaging/MacOS/OpenAFS.info.in
+++ b/src/packaging/MacOS/OpenAFS.info.in
@@ -1,5 +1,5 @@
 Title OpenAFS
-Version @VERSION@
+Version @PACKAGE_VERSION@
 Description The OpenAFS distributed filesystem. This package installs a ready-to-run client for OpenAFS. See http://www.openafs.org for more information.
 DefaultLocation /
 Diskname (null)
diff --git a/src/packaging/MacOS/buildpkg.sh.in b/src/packaging/MacOS/buildpkg.sh.in
index 143bda7..ec3711c 100644
--- a/src/packaging/MacOS/buildpkg.sh.in
+++ b/src/packaging/MacOS/buildpkg.sh.in
@@ -306,14 +306,14 @@ if [ $secondpass = yes ]; then
     if [ $majorvers -ge 8 ]; then
 	mv $CURDIR/OpenAFS-debug-extension.pkg $CURDIR/dmg
     fi
-    rm -rf $CURDIR/OpenAFS-@VERSION@-$RELNAME.dmg
+    rm -rf $CURDIR/OpenAFS-@PACKAGE_VERSION@-$RELNAME.dmg
     cp $RESSRC/Uninstall $CURDIR/dmg/Uninstall.command
     cp $RESSRC/DS_Store $CURDIR/dmg/.DS_Store
     mkdir $CURDIR/dmg/.background
     cp $RESSRC/afslogo.jpg $CURDIR/dmg/.background
 #    hdiutil create -srcfolder $CURDIR/dmg -volname OpenAFS -anyowners $CURDIR/OpenAFS.dmg
     hdiutil makehybrid -hfs -hfs-volume-name OpenAFS -hfs-openfolder $CURDIR/dmg $CURDIR/dmg -o $CURDIR/TMP.dmg
-    hdiutil convert -format UDZO $CURDIR/TMP.dmg -o $CURDIR/OpenAFS-@VERSION@-$RELNAME.dmg
+    hdiutil convert -format UDZO $CURDIR/TMP.dmg -o $CURDIR/OpenAFS-@PACKAGE_VERSION@-$RELNAME.dmg
     rm $CURDIR/TMP.dmg
     rm -rf $CURDIR/dmg
     # Unfortunately, sudo sets $USER to root, so I can't chown the 
diff --git a/src/packaging/MacOS/pkgbuild.sh.in b/src/packaging/MacOS/pkgbuild.sh.in
index 3570d7a..d65f585 100644
--- a/src/packaging/MacOS/pkgbuild.sh.in
+++ b/src/packaging/MacOS/pkgbuild.sh.in
@@ -426,7 +426,7 @@ if [ x"$PASS2" = x1 ]; then
 
     rm -rf "$CURDIR"/OpenAFS-debug-extension.pkg
     /usr/bin/pkgbuild --root "$DPKGROOT" --id org.openafs.OpenAFS-debug.pkg \
-		      --version '@VERSION@' \
+		      --version '@PACKAGE_VERSION@' \
 	     "$CURDIR"/OpenAFS-debug-extension.pkg
 
     cd "$RESSRC"
@@ -445,7 +445,7 @@ if [ x"$PASS2" = x1 ]; then
 
     rm -rf "$CURDIR"/OpenAFS-dist.pkg
     /usr/bin/pkgbuild --root "$PKGROOT" --id org.openafs.OpenAFS.pkg \
-		      --version '@VERSION@' \
+		      --version '@PACKAGE_VERSION@' \
 		      --scripts "$PKGRES" "$CURDIR"/OpenAFS-dist.pkg
 
     rm -rf "$PKGRES"
@@ -466,7 +466,7 @@ if [ x"$PASS2" = x1 ]; then
     sed -e "s/%%OSX_MAJOR_CUR%%/$THISREL/g" \
 	-e "s/%%OSX_MAJOR_NEXT%%/$(( $THISREL + 1 ))/g" \
 	-e "s,%%PRES_EXTRA%%,$PRES_EXTRA,g" \
-	-e "s/%%OPENAFS_VERSION%%/@VERSION@/g" \
+	-e "s/%%OPENAFS_VERSION%%/@PACKAGE_VERSION@/g" \
 	< Distribution.xml.in > Distribution.xml
 
     rm -rf "$CURDIR/prod"
@@ -484,7 +484,7 @@ if [ x"$PASS2" = x1 ]; then
 
     rm -rf "$CURDIR/dmg"
     rm -f "$CURDIR/TMP.dmg"
-    rm -rf "$CURDIR/OpenAFS-@VERSION@-$RELNAME.dmg"
+    rm -rf "$CURDIR/OpenAFS-@PACKAGE_VERSION@-$RELNAME.dmg"
 
     mkdir "$CURDIR"/dmg
     mv "$CURDIR"/OpenAFS.pkg "$CURDIR"/dmg
@@ -500,8 +500,8 @@ if [ x"$PASS2" = x1 ]; then
     hdiutil create -srcfolder "$CURDIR"/dmg -volname OpenAFS \
 	    -o "$CURDIR"/TMP.dmg
     hdiutil convert -format UDZO "$CURDIR"/TMP.dmg \
-	    -o "$CURDIR/OpenAFS-@VERSION@-$RELNAME".dmg
+	    -o "$CURDIR/OpenAFS-@PACKAGE_VERSION@-$RELNAME".dmg
 
     echo
-    echo "Created $CURDIR/OpenAFS-@VERSION@-$RELNAME".dmg
+    echo "Created $CURDIR/OpenAFS-@PACKAGE_VERSION@-$RELNAME".dmg
 fi
diff --git a/src/packaging/RedHat/openafs.spec.in b/src/packaging/RedHat/openafs.spec.in
index 5282c4a..d0298d0 100644
--- a/src/packaging/RedHat/openafs.spec.in
+++ b/src/packaging/RedHat/openafs.spec.in
@@ -1,6 +1,6 @@
 # Openafs Spec $Revision$
 
-%define afsvers @VERSION@
+%define afsvers @PACKAGE_VERSION@
 %define pkgvers @LINUX_PKGVER@
 # for beta/rc releases make pkgrel 0.<tag>
 # for real releases make pkgrel 1 (or more for extra releases)
diff --git a/src/tests/run-tests.in b/src/tests/run-tests.in
index d0fc73d..1ab12ad 100755
--- a/src/tests/run-tests.in
+++ b/src/tests/run-tests.in
@@ -28,8 +28,8 @@ srcdir=@srcdir@
 objdir=`pwd`
 SHELL=/bin/sh
 SHELLVERBOSE=
-VERSION=@VERSION@
-PACKAGE=@PACKAGE@
+VERSION=@PACKAGE_VERSION@
+PACKAGE=@PACKAGE_TARNAME@
 host=@host@
 RUNAS=
 FS=${AFSSRVBINDIR}/fs
